src/commands/deploy/labels.js

```diff
@@ -35,10 +36,10 @@ const upgradeService = ({ projectId, id, deploymentConfig, containers }, image)
 module.exports = async ({ key, value, image, namespace }) => {
   log('Retrieving workloads...');
   const namespaces = await getNamespaces(namespace);
-  const promised = await Promise.all(namespaces.map(n => getWorkloads(n, key, value)));;
-  const workloads = promised.reduce((arr, w) => arr.concat(w), []);
-
-  log(`Upgrading ${workloads.length} workloads...`);
-  await Promise.all(workloads.map(async w => await upgradeService(w, image)));
+  async.mapLimit(namespaces, 2, async function(namespace) {
+    const promised = await getWorkloads(namespace, key, value);
+    const workloads = promised.reduce((arr, w) => arr.concat(w), []);
+    await Promise.all(workloads.map(async w => await upgradeService(w, image)));
+  });
```

Comment:
If you want to make this more efficient, await the namespace and workload queries first, then pass them into the getWorkloads and getNamespaces methods (from 3n to n+2).

I would rather not include the async library if possible -- all mapLimit is doing is chunking your calls into arrays of `limit` length, and then sequentially running `await Promise.all(chunk)`;

You can replicate this yourself without an extra library by doing the same, or just running them sequentially:
```js
const main = async (namespace) => {
  /** upgradeService logic */
};

namespaces
  .reduce((c, k, i) => c.then(res => main(k).then(r => [ ...res, r ])), Promise.resolve([]))
  .then(() => log('All done!'));
```

I'd prefer sequential execution over another library.